packages/core-data/src/reducer.js

```diff
@@ -99,6 +99,23 @@ export function themeSupports( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing Upload permissions.
+ *
+ * @param  {Object}  state  Current state.
+ * @param  {Object}  action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function hasUploadPermissions( state = {}, action ) {
```

Comment:
I don't think an empty object is intended as the default value here. I'm assuming in current usage it will be inferred as effectively truthy (since `Boolean( {} ) === true`). Should then the default value be `true` instead? In any case, probably should be a boolean at least.